services/monitor/src/ChainMonitor.ts

```diff
@@ -365,6 +381,36 @@ export default class ChainMonitor extends EventEmitter {
     }
   };
 
+  private async triggerSimilarityVerificationFallback(
+    creatorTxHash: string,
+    address: string,
+  ) {
+    if (!this.similarityVerificationClient) {
+      this.chainLogger.debug("Similarity search disabled, skipping fallback", {
+        address,
+      });
+      return;
+    }
+
+    this.chainLogger.info("Triggering similarity search fallback", {
```

Comment:
This whole section seems unnecessary and can be handled inside the SimilarityVerificationClient. Then here we can simple call `this.client.trigger()`